src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -16,58 +14,66 @@ export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrat
         const {
             id: paymentId,
-            initializationData: { clientId, intent, isPayPalCreditAvailable },
+            initializationData,
         } = state.paymentMethods.getPaymentMethodOrThrow(options.methodId);
-        const paypalOptions = options.paypalCommerce;
 
-        if (!clientId) {
+        if (!initializationData.clientId) {
             throw new InvalidArgumentError();
         }
 
-        return this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout())
-            .then(state => {
-                const cart = state.cart.getCart();
-
-                if (!cart) {
-                    throw new MissingDataError(MissingDataErrorType.MissingCart);
-                }
-
-                const buttonParams: ButtonsOptions = {
-                    createOrder: () => this._setupPayment(cart),
-                    onApprove: data => this._tokenizePayment(paymentId, data),
-                };
-
-                if (paypalOptions && paypalOptions.style) {
-                    buttonParams.style = this._validateStyleParams(paypalOptions.style);
-                }
-
-                const paramsScript: PaypalCommerceScriptOptions = {
-                    clientId,
-                    intent,
-                    currency: cart.currency.code,
-                    commit: false,
-                };
-
-                if (!isPayPalCreditAvailable) {
-                    paramsScript.disableFunding = 'credit';
-                }
-
-                return this._paypalScriptLoader.loadPaypalCommerce(paramsScript)
-                    .then(paypal => paypal.Buttons(buttonParams).render(`#${options.containerId}`));
-            });
+        state = await this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout());
+        const cart = state.cart.getCart();
+        const paypalOptions = options.paypalCommerce;
+
+        if (!cart) {
+            throw new MissingDataError(MissingDataErrorType.MissingCart);
+        }
+
+        const buttonParams: ButtonsOptions = {
+            createOrder: async () => this._setupPayment(cart.id),
+            onApprove: data => this._tokenizePayment(paymentId, data),
+        };
+
+        if (paypalOptions && paypalOptions.style) {
+            buttonParams.style = this._validateStyleParams(paypalOptions.style);
+        }
+
+        const paramsScript = this._getParamsScript(initializationData, cart);
+        const paypal = await this._paypalScriptLoader.loadPaypalCommerce(paramsScript);
+
+        return paypal.Buttons(buttonParams).render(`#${options.containerId}`);
     }
 
     deinitialize(): Promise<void> {
         return Promise.resolve();
     }
 
+    private async _setupPayment(cartId: string): Promise<string> {
+        const res = await this._paypalCommerceRequestSender.setupPayment(cartId);
+
+        return res.orderId;
+    }
+
+    private _tokenizePayment(paymentId: string, data: ApproveDataOptions) {
```

Comment:
you could this instead (sorry for not picking it up before)
```js
private _tokenizePayment(paymentId: string, { orderId }: ApproveDataOptions) {
  if (!orderID) {
    // ..
  }
```

Reply:
Thank you!